### PR TITLE
More simplifications

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Still more simplifications

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ## Optimizations
 
+- Added more rules for simplifying expressions, especially around Concatenations. Also, meshes constructed from multiple domains are now memoized ([#2443](https://github.com/pybamm-team/PyBaMM/pull/2443))
```
Comment (brosaplanella) [inline CHANGELOG.md:9]:
```suggestion
- Added more rules for simplifying expressions, especially around Concatenations. Also, meshes constructed from multiple domains are now memorized ([#2443](https://github.com/pybamm-team/PyBaMM/pull/2443))
```

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (tinosulzer) [reply CHANGELOG.md:9]: This refers to memoization https://en.wikipedia.org/wiki/Memoization, but I will change it to "cached"

Comment (brosaplanella) [reply CHANGELOG.md]: Oops, didn't know that, sorry! If it is a word, happy to keep it as it is.

**Pull request merged**